pyproject.toml

```
@@ -138,9 +140,17 @@ lint.allowed-confusables = [
 lint.ignore = [
     "COM812", # missing-trailing-comma (ruff-format recommended)
     "D203", # one-blank-line-before-class
+    "D205",
     "D212", # blank-line-before-class
+    "D401",
+    "EM101",
+    "EM102",
     "ISC001", # single-line-implicit-string-concatenation (ruff-format recommended)
+    "PLR0912",
+    "PLR0913",
```

These 2 rules are useful as well, but I am happy to ignore them if a lot of functions are violating them and we can't do anything about it.
```suggestion
    "PLR0912",  # too-many-branches
    "PLR0913",  # too-many-arguments
```

---

Could you explain how they are useful? I genuinely don't understand the point. If a code can be written better (e.g. parts can be factored out into standalone functions) then that's going to be done in code review. AFAICT these too are just producing noise for numerical functions.